src/main/java/co/cask/bigquery/BigQuerySource.java

```diff
@@ -237,6 +237,7 @@ public Schema getSchema(Request request) throws Exception {
       for (Field field : bgSchema.getFields()) {
         Field.Type type = field.getType();
         Schema.Type cType = Schema.Type.STRING;
```

Comment:
this existing logic is incorrect, we can't just assume a string. There should be an else clause at the end of the else-ifs that throws an exception about an unsupported type.